### PR TITLE
Fully functional version of Job Builder Details is working

### LINKED ISSUE #862: Job Poster Builder: Step 02 - Job Information
# Description
In this step, the hiring manager will provide basic job information like Job Title, Term, Classification, Level, Security Level, Language Profile, Location and are Remote Work, Telework and Flexible Hours allowed. After this step is complete we also show an encouraging render of some of the data.

As we add inputs to this UI, we will be removing the corresponding ones from the old UI that admins are still using (/en/manager/jobs/##/edit).

- [x] This task requires a unit test.

**Acceptance Criteria**
***Page & Nav***
- [ ] Depending on the button pressed on Step 0 (#942), we should either be on the /en version of the page or on the /fr version.
- [ ] If we are on the /en version of the page, all the "translatable" fields (like job title) should only be showing the English Input (and French only for /fr). This means you will need to toggle between the language of the page if you ever want to complete both languages.
- [ ] This page is the second on the progress tracker (top right) and should follow the same logic as described in #942

***Form***
- [ ] Job Title is the input for the EN or FR job title based on the language of the page you're on. This input should be removed from the old form (as all inputs we're moving to this form should be).
- [ ] "How long is the term (in months)" corresponds to Position Duration in the Timetable section of the old form.

Classification and Level are NEW. Previously this was a single text input with some regex restrictions. Now they have separate dropdowns.
- [ ] "What is the classification" should ideally get saved to the old text field we had for Classification. That will help us maintain backwards compatibility with older job posters.
- [ ] Classification should be removed from the old form.
- [ ] "What is the level" should be saved to a new integer field
- [ ] The Level pull-down is simply a list from 1 - 9
- [ ] The job poster needs to get updated to display this information correctly. It should display "Classification"-"Level". For example CS-04 or ENsur-05.

- [ ] "What is the Security Level?" replaces "Security Clearance" from the old form
- [ ] "What is the language profile" replaces "Language Requirement" from the old form.
- [ ] The text for "language profile" we want to display for managers is different then the text to display to candidates. Please see content doc for how they should appear here.
- [ ] Only NOC code should remain under "classifications" in the old form.

***Remote Work***
- [ ] As in the mock-up there is some added text below the input label.
- [ ] Remote work now has several options. On the old form it was just a checkbox. This new input should replace the old one.
- [ ] To keep thing simple, let's just map the two new "Yes" options to true and the "No" option to false. That way we don't need to make changes to the job poster. We have created an issue for the improved version (#949) but it is not prioritized for now.

***Telework and Flexible Hours***
These fields currently exist on the Manager Profile "/en/manager/profile/##/edit". 
- [ ] As in the mock-up there is some added text below the input label.
- [ ] Create new fields on the job poster model to capture this data (Gray is reaching beyond his level of expertise here, please confirm with other devs if needed)
- [ ] Removing these fields from the manager profile will be tricky. Ideally we copy the current values in every manager's profile over to the new fields of all their job posters. But feel free to propose other solutions.

***Validation***
- [ ] All fields are required.

**Job Poster Card**
When the Next button is clicked, the next page shows the Job Card with some encouraging text. The Job Poster Card will display the following details pulled from the previous two pages;

- [ ] Job title 
- [ ] Department
- [ ] Location
- [ ] Duration or Term (in months)
- [ ] Is Remote Work Allowed (Yes or No)

- [ ] Additional information such as Average Annual Salary, Apply by and Applicants to Date will be posted on the preview but will be static only content.

Functionality on the Job Poster Card will include;
- [ ] a link back to previous screen so that users can make changes, 
- [ ] a save or continue button to move to the next step,
- [ ] a rollover on the 'View Job' button informing users that 'There's nothing to view yet but, we're off to a good start'.

**Link to Wireframe(s):**
[Job Details and Preview](https://xd.adobe.com/view/19e2b905-9ef3-4ba3-44b7-b5317334e680-6fd3/)

1. Click each pull-down menu to see available options. 
2. Click next a couple times to get to the Job Card

**Link to Content:**
[Job Details and Preview](https://docs.google.com/document/d/1hoHWlctAIVbFHmyzBmnGyaRjvkBIcvYkMxSVzmiLm2Q/edit?usp=sharing)

***Components:***
- [ ] UI
- [ ] Step 1 React Component - connected to redux
- [ ] Step 1 React localization
- [ ] Step 1 Preview UI
- [ ] Step 1 Preview Component - connected to redux
- [ ] Step 1 Preview localization

### PR DESCRIPTION
**Pull request by tristan-orourke**
That is, it is fully integrated with backend server. It can be accessed at the url `/manager/jobs/{jobId}/builder/details`, or `manager/job-builder/details` to start a new job. 
When you click next, if the form is error free a version will be saved to the server. If the server returns an error, the user is notified. If the save is successful, a preview appears in the modal. For now, clicking `next` in the modal simply returns you to the jobs index.

Resolves #862

### CODE REVIEW

File: resources/views/manager/job-builder-details.html.twig [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    <a id="skipLink"></a>
+
+    <div
+        data-clone=""
+        id="job-builder-details"
+        data-job-id="{{jobId}}"
+    >
+    </div>
+
+    <script async defer
```
Comment (Khristophor) [inline resources/views/manager/job-builder-details.html.twig]: Good call, but should put this in the `scripts` block that's in the base layout template.

File: app/Http/Controllers/Api/JobApiController.php [hunk 2]
```diff
@@ -91,7 +90,8 @@ public function update(UpdateJobPoster $request, JobPoster $job)
         // and returned by UpdateJobPoster->validatedData(), will be set.
         $job->fill($data);
         $job->save();
-        return $this->jobToArray($job);
+        // return response()->json($request->input());
```
Comment (Khristophor) [inline app/Http/Controllers/Api/JobApiController.php]: Assume this is safe to remove

File: app/Services/WhichPortal.php [hunk 3]
```diff
@@ -3,16 +3,21 @@
...
 
-    public function isManagerPortal() {
-        $routeName = Route::currentRouteName();
-        return str_is('manager.*',$routeName);
+    public function isManagerPortal()
+    {
+        $url = URL::current();
+        return str_is('*/manager/*', $url);
+        // $routeName = Route::currentRouteName();
+        // return str_is('manager.*', $routeName);
```
Comment (Khristophor) [inline app/Services/WhichPortal.php]: Assume these are safe to remove

File: resources/assets/js/ReduxApp.tsx [hunk 4]
```diff
@@ -41,7 +43,10 @@ const IntlRouteContainer: React.FunctionComponent<
   const { locale } = match.params;
   return (
     <IntlContainer locale={locale}>
-      <Route path={`${match.path}/manager`} component={ManagerPortal} />
+      <>
+        <ErrorToast />
```
Comment (Khristophor) [inline resources/assets/js/ReduxApp.tsx]: Good call moving this up and out of the assessment plan 👍

File: resources/assets/js/components/JobDetails/JobDetails.tsx [hunk 5]
```diff
@@ -253,44 +327,34 @@ const JobDetails: React.FunctionComponent<
       )
       .required(intl.formatMessage(validationMessages.required)),
     level: Yup.number()
-      .min(1, intl.formatMessage(validationMessages.invalidSelection))
+      .min(1)
```
Comment (Khristophor) [inline resources/assets/js/components/JobDetails/JobDetails.tsx]: Any particular reason the message was removed from this one?

File: resources/assets/js/components/JobDetails/JobDetails.tsx [hunk 6]
```diff
@@ -334,13 +402,21 @@ const JobDetails: React.FunctionComponent<
...
         <Formik
+          enableReinitialize
           initialValues={initialValues}
           validationSchema={jobSchema}
           onSubmit={(values, actions): void => {
             // The following only triggers after validations pass
-            setIsModalVisible(true);
-            handleSubmit(values);
-            actions.setSubmitting(false); // Required by Formik to finish the submission cycle
+            // setIsModalVisible(true);
```
Comment (Khristophor) [inline resources/assets/js/components/JobDetails/JobDetails.tsx]: Assume this is safe to remove

File: resources/assets/js/components/JobDetails/JobDetailsPage.tsx [hunk 7]
```diff
@@ -0,0 +1,145 @@
...
+} => ({
+  loadJob: (jobId: number): void => {
+    dispatch(fetchJob(jobId));
+    dispatch(setSelectedJob(jobId));
+  },
+  handleCreateJob: async (newJob: Job): Promise<boolean> => {
+    const result = await dispatch(createJob(newJob));
+    if (!result.error) {
+      console.log("Handling create job result");
+      // result.payload.then(resultJob => {
```
Comment (Khristophor) [inline resources/assets/js/components/JobDetails/JobDetailsPage.tsx]: Is this supposed to be commented?

File: resources/assets/js/components/JobDetails/JobDetailsPage.tsx [hunk 8]
```diff
@@ -0,0 +1,145 @@
...
+        <div
+          data-c-container="form"
+          data-c-padding="top(triple) bottom(triple)"
+        >
+          <h3
+            data-c-font-size="h3"
+            data-c-font-weight="bold"
+            data-c-margin="bottom(double)"
+          >
+            Job Loading...
```
Comment (Khristophor) [inline resources/assets/js/components/JobDetails/JobDetailsPage.tsx]: We'll need to localize this

File: resources/assets/js/store/asyncAction.ts [hunk 9]
```diff
@@ -8,6 +8,7 @@ import {
   getJSON,
 } from "redux-api-middleware"; // RSAA = '@@redux-api-middleware/RSAA'
 import { ErrorAction } from "./createAction";
+import moment = require("moment");
```
Comment (Khristophor) [inline resources/assets/js/store/asyncAction.ts]: 😢

File: routes/web.php [hunk 10]
```diff
@@ -1,6 +1,7 @@
 <?php
 
 use Illuminate\Support\Facades\Lang;
+use Illuminate\Http\Request;
```
Comment (Khristophor) [inline routes/web.php]: Curious why this is added?

File: resources/assets/js/components/JobDetails/JobDetails.tsx (hunk 5 above)
Comment (tristan-orourke) [reply resources/assets/js/components/JobDetails/JobDetails.tsx]: Nope! Not sure how that happened.

File: resources/assets/js/components/JobDetails/JobDetailsPage.tsx (hunk 7 above)
Comment (tristan-orourke) [reply resources/assets/js/components/JobDetails/JobDetailsPage.tsx]: Its supposed to be deleted, I did the same thing with await instead.

File: resources/assets/js/components/JobDetails/JobDetailsPage.tsx (hunk 8 above)
Comment (tristan-orourke) [reply resources/assets/js/components/JobDetails/JobDetailsPage.tsx]: It needs more ui love than just that. But localizing is easy.

File: resources/assets/js/store/asyncAction.ts (hunk 9 above)
Comment (tristan-orourke) [reply resources/assets/js/store/asyncAction.ts]: Quality sprint, right? ...Right?

File: routes/web.php (hunk 10 above)
Comment (tristan-orourke) [reply routes/web.php]: Ah, left over from debugging.

File: resources/assets/js/store/asyncAction.ts (hunk 9 above)
Comment (tristan-orourke) [reply resources/assets/js/store/asyncAction.ts]: But I fixed the syntax here at least. `import moment from "moment";`

**Pull request merged**